tests/foreman/api/test_errata.py

```diff
@@ -49,872 +49,881 @@
+                'included': {'ids': host_ids},
+                'content_type': 'package',
+                'content': [package_name],
             }
         )
-
-    def _install_package(
-        self, clients, host_ids, package_name, via_ssh=True, rpm_package_name=None
-    ):
-        """Install package via SSH CLI if via_ssh is True, otherwise
-        install via http api: PUT /api/v2/hosts/bulk/install_content
-        """
-        if via_ssh:
-            for client in clients:
-                result = client.run(f'yum install -y {package_name}')
-                self.assertEqual(result.return_code, 0)
-                result = client.run(f'rpm -q {package_name}')
-                self.assertEqual(result.return_code, 0)
-        else:
-            entities.Host().install_content(
-                data={
-                    'organization_id': self.org.id,
-                    'included': {'ids': host_ids},
-                    'content_type': 'package',
-                    'content': [package_name],
-                }
-            )
-            self._validate_package_installed(clients, rpm_package_name)
-
-    def _validate_package_installed(
-        self, hosts, package_name, expected_installed=True, timeout=120
-    ):
-        """Check whether package was installed on the list of hosts."""
-        for host in hosts:
-            for _ in range(timeout // 15):
-                result = host.run(f'rpm -q {package_name}')
-                if (
-                    result.return_code == 0
-                    and expected_installed
-                    or result.return_code != 0
-                    and not expected_installed
-                ):
-                    break
-                sleep(15)
-            else:
-                self.fail(
-                    'Package {} was not {} host {}'.format(
-                        package_name,
-                        'installed on' if expected_installed else 'removed from',
-                        host.hostname,
-                    )
-                )
-
-    def _validate_errata_counts(self, host, errata_type, expected_value, timeout=120):
-        """Check whether host contains expected errata counts."""
-        for _ in range(timeout // 5):
-            host = host.read()
-            if host.content_facet_attributes['errata_counts'][errata_type] == expected_value:
+        _validate_package_installed(clients, rpm_package_name)
+
+
+def _validate_package_installed(hosts, package_name, expected_installed=True, timeout=120):
+    """Check whether package was installed on the list of hosts."""
+    for host in hosts:
+        for _ in range(timeout // 15):
+            result = host.run(f'rpm -q {package_name}')
+            if (
+                result.return_code == 0
+                and expected_installed
+                or result.return_code != 0
+                and not expected_installed
+            ):
                 break
-            sleep(5)
+            sleep(15)
```

Comment:
TODO, identify a condition and use wait_for instead of static sleeps.

Reply:
https://github.com/SatelliteQE/robottelo/issues/8403

BTW, I did not change that, that diff is misleading, its from line 138 in the pre-change file.